src/elements/Pagination/index.jsx

```diff
@@ -0,0 +1,236 @@
+    let rightLength = maxPaginationSize - leftLength - 1
+
+    if ((totalPages - currentPage) < pivot) {
+      rightLength = totalPages - currentPage
+      leftLength = maxPaginationSize - rightLength - 1
+    }
+
+    if (maxPaginationSize >= totalPages) {
+      for (let i = 0; i < totalPages; i += 1) {
+        steps.push(i + 1)
+      }
+    } else {
+      steps = this.calculatePreviousSteps(leftLength, currentPage)
+        .concat(currentPage, this.calculateFollowingSteps(rightLength, currentPage, totalPages))
+    }
+    return steps
+  }
+
+  calculatePreviousSteps (leftLength, currentPage) {
+    const res = []
+
+    if ((currentPage - 1) > leftLength) {
+      for (let i = 0; i < leftLength; i += 1) {
+        if (i === leftLength - 1) {
+          res.unshift(1)
+        } else if (i === leftLength - 2) {
+          res.unshift('ellipsis')
+        } else {
+          res.unshift(currentPage - i - 1)
+        }
+      }
+    } else {
+      for (let i = 0; i < leftLength; i += 1) {
+        res.push(i + 1)
+      }
+    }
+    return res;
+  }
+
+  calculateFollowingSteps (rightLength, currentPage, totalPages) {
+    const res = []
+
+    if ((totalPages - currentPage) > rightLength) {
+      for (let i = 0; i < rightLength; i += 1) {
+        if (i === rightLength - 1) {
+          res.push(totalPages)
+        } else if (i === rightLength - 2) {
+          res.push('ellipsis')
+        } else {
+          res.push(currentPage + i + 1)
+        }
+      }
+    } else {
+      for (let i = 0; i < rightLength; i += 1) {
+        res.push(currentPage + i + 1)
+      }
+    }
+    return res;
+  }
+
+  renderPagination () {
+    const { currentPage, totalPages } = this.state
+    if (currentPage < 1) {
+      this.changePage(1)
+    } else if (currentPage > totalPages) {
+      this.changePage(totalPages)
+    }
+
+    const paginations = []
+    const steps = this.calculateSteps();
+
+    for (let i = 0; i < steps.length; i += 1) {
+      const label = steps[i];
+      let isClickable = true;
+
+      if (currentPage === steps[i]) {
+        isClickable = false;
+      }
+      if (label === 'ellipsis') {
+        paginations.push(<PaginationLink isPageBreak={true} key={i} />)
+      } else {
+        paginations.push(
+          <PaginationLink
+            label={label}
+            isClickable={isClickable}
+            onClick={(page) => { this.onClick(page) }}
+            key={i} />
+        )
+      }
+    }
+
+    return paginations;
+  }
+
+  renderLeftCaret () {
+    const isDisabled = this.state.currentPage <= 1
+    const iconElement = this.getCaretIcon('left', isDisabled)
+
+    return (
+      <li className={classNames('Pagination__caret', 'Pagination__caret--left')}>
+        { isDisabled
+          ? <span>{iconElement}</span>
+          : <a onClick={() => {
+            this.onClick(this.state.currentPage - 1)
+          }}>
+            {iconElement}
+          </a> }
+      </li>
+    )
+  }
```

Comment:
I think you thought too much on this icon, so the code got complicated unnecessarily. I think this should be enough. I don't think we need `getCaretIcon` method.
```jsx
static defaultProps = {
  currentPage: 1,
  maxPaginationSize: 7,
  fontSize: 'medium',
  icon: {
    type: 'githubOcticons',
    name: 'GoTriangle',
    size: 14,
  }
}
```
```jsx
renderLeftCaret () {
  const { icon } = this.props
  const isDisabled = this.state.currentPage <= 1
  
  return (
    <li className={classNames('Pagination__caret', 'Pagination__caret--left')}>
    {  isDisabled
      ? <Icon type={icon.type} name={`${icon.name}Left`} size={icon.size} color={disabled} />
      : <a onClick={() => this.onClick(this.state.currentPage + 1)}>
        <Icon type={icon.type} name={`${icon.name}Left`} size={icon.size} color={icon.color} />
      </a>
    </li>
  )
}
```